cmd/skopeo/copy.go

```diff
@@ -72,16 +73,36 @@ See skopeo(1) section "IMAGE NAMES" for the expected format
 	flags.StringSliceVar(&opts.additionalTags, "additional-tag", []string{}, "additional tags (supports docker-archive)")
 	flags.BoolVarP(&opts.quiet, "quiet", "q", false, "Suppress output information when copying images")
 	flags.BoolVarP(&opts.all, "all", "a", false, "Copy all images if SOURCE-IMAGE is a list")
+	flags.StringVar(&opts.listProcessing, "list-processing", "CopySystemImage", "How to process image lists. Choose one of CopySystemImage, CopyAllImages, or CopyNoImages")
 	flags.BoolVar(&opts.removeSignatures, "remove-signatures", false, "Do not copy signatures from SOURCE-IMAGE")
 	flags.StringVar(&opts.signByFingerprint, "sign-by", "", "Sign the image using a GPG key with the specified `FINGERPRINT`")
 	flags.StringVar(&opts.digestFile, "digestfile", "", "Write the digest of the pushed image to the specified file")
 	flags.VarP(commonFlag.NewOptionalStringValue(&opts.format), "format", "f", `MANIFEST TYPE (oci, v2s1, or v2s2) to use in the destination (default is manifest type of source, with fallbacks)`)
 	flags.StringSliceVar(&opts.encryptionKeys, "encryption-key", []string{}, "*Experimental* key with the encryption protocol to use needed to encrypt the image (e.g. jwe:/path/to/key.pem)")
 	flags.IntSliceVar(&opts.encryptLayer, "encrypt-layer", []int{}, "*Experimental* the 0-indexed layer indices, with support for negative indexing (e.g. 0 is the first layer, -1 is the last layer)")
 	flags.StringSliceVar(&opts.decryptionKeys, "decryption-key", []string{}, "*Experimental* key needed to decrypt the image")
+	flags.MarkDeprecated("all", "please use --list-processing=CopyAllImages instead")
 	return cmd
 }
 
+// parseListProcessing parses the list processing selection
+// It returns the copy.ImageListSelection to use with image.Copy option
+func parseListProcessing(listProcessing string) (copy.ImageListSelection, error) {
+	switch listProcessing {
+	case "CopySystemImage":
+		return copy.CopySystemImage, nil
+	case "CopyAllImages":
+		return copy.CopyAllImages, nil
+	// There is no CopyNoImages value in copy.ImageListSelection, but because we
+	// don't provide an option to select a set of images to copy, we can use
+	// CopySpecificImages as CopyNoImages.
+	case "CopySpecificImages", "CopyNoImages":
```

Comment:
Providing `CopySpecificImages` without a way to set the list doesn’t seem useful. We can just not implement it, comment here that it is not supported, and if someone ends up needing it (I’m not quite sure that will ever happen, BTW — it seems more useful when integrated into a complex tool in Go), we can add it later.

Reply:
Yes, I was just trying to match up the options from image, but really you're right - this should be skopeo UX not image internals.

Reply:
Non-blocking: Maybe just add a comment above the `default`:
> We don't expose CopySpecificImages other than index-only above, because we currently don't provide an option to choose the images to copy.  That could be added in the future.

This PR is already a good enough record of that thought process; a comment directly in the code would make it a bit easier to find.